client/src/pages/CodeUpload.js

```diff
@@ -0,0 +1,200 @@
+  const startSubmit = () => {
+    let languageError = requestLanguage === "";
+    let titleError = requestTitle === "";
+    let errorSet = new Set();
+    if (languageError || titleError || !editorHasContent) {
+      if (languageError) {
+        errorSet.add(
+          "Please select a language for your request before submitting"
+        );
+      }
+      if (titleError) {
+        errorSet.add("Please add a title for your request before submitting");
+      }
+      if (!editorHasContent) {
+        errorSet.add("Please add some content to your review request");
+      }
+      setPageAlerts(errorSet);
+      setAlertState(true);
+    } else {
+      //No errors, begin submitting
+      setPageAlerts(new Set());
+      setSubmitState(true);
+    }
+  };
+
+  //reset alerts
+  const resetAlerts = () => {
+    setAlertState(false);
+  };
+
+  //check if user has typed into editor
+  const handleHasContent = value => {
+    setEditorHasContent(value);
+  };
+
+  //get data from editor component
+  const handleSubmit = async data => {
+    //Wrap up editor data with user and language data and send to server.
+    const requestData = {
+      title: requestTitle,
+      language: requestLanguage,
+      content: data,
+      user: testUser
+    };
+
+    try {
+      const response = await axios({
+        method: "post",
+        url: `/${testUser._id}/create-request`,
+        data: {
+          dataObj: JSON.stringify(requestData)
+        }
+      });
+      const success = response.success;
+      if (success) {
+        //redirect user to their reviews page
+        //success snackbar?
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  return (
+    <div className={classes.root}>
+      <Grid className={classes.wrapper} container justify="center" spacing={2}>
+        <Typography className={classes.header} variant="h2" align="center">
+          Request a Code Review
+        </Typography>
+        <Grid item xs={8}>
+          <TextField
+            className={classes.textInput}
+            align="left"
+            label="Title"
+            variant="outlined"
+            onChange={handleTitleChange}
+          />
+        </Grid>
+        <Grid item xs={4}>
+          <InputLabel id="language-select-label">Language</InputLabel>
+          <Select
+            className={classes.languageSelect}
+            labelid="language-select-label"
+            id="language-select"
+            value={requestLanguage}
+            onChange={handleLanguageChange}
+          >
+            {getLanguages.map(language => {
+              return (
+                <MenuItem value={languageGrammar[language]} key={language}>
+                  {language}
+                </MenuItem>
+              );
+            })}
+          </Select>
+        </Grid>
+        <Grid item xs={12}>
+          <TextEditor
+            selectedLanguage={requestLanguage}
+            onSubmit={handleSubmit}
+            didSubmit={submitState}
```

Comment:
The flow of submitting is confusing and hard to follow, it could probably be a bit better designed